clustpy/deep/_utils.py

```diff
@@ -94,8 +94,29 @@ def detect_device(device: torch.device = None) -> torch.device:
     return device
 
 
-def encode_batchwise(dataloader: torch.utils.data.DataLoader, module: torch.nn.Module,
-                     device: torch.device) -> np.ndarray:
+def get_device_from_module(module: torch.nn.Module) -> torch.device:
```

Comment:
Added this method to get the device of a torch model. Therefore, when encoding/decoding data we dont have to specify the device of the data anymore (must be the same as for the model)

Reply:
sounds goods. Does this fix the problem of clustpy of always choosing gpu:0 on multi-gpu systems as well? Because this was a problem I encountered quite often

Reply:
Added an input parameter for choosing a specific GPU to all deep clustering algorithms